Assets/Scripts/Player/PlayerController.cs

```diff
@@ -47,4 +58,25 @@ protected virtual void ShootWeapon()
             weapon.Fire(transform.position, Camera.main.ScreenToWorldPoint(Input.mousePosition));
         }
     }
+
+    private void FlipCharacterAccordingToMousePosition(Vector2 mousePosition)
+    {
+        float mousePosBasedOnCharPos = mousePosition.x - transform.position.x;
+
+        if (!charIsFacingRight && mousePosBasedOnCharPos > 0)
+        {
+            Vector3 newAngles = transform.eulerAngles;
+            newAngles.y = 0;
+            transform.eulerAngles = newAngles;
+
+            charIsFacingRight = true;
+        }
+        else if (charIsFacingRight && mousePosBasedOnCharPos < 0)
+        {
+            Vector3 newAngles = transform.eulerAngles;
+            newAngles.y = 180;
+            transform.eulerAngles = newAngles;
+            charIsFacingRight = false;
+        }
+    }
```

Comment:
Is "charIsFacingRight" bool required. Can't we make it only with the "mousePosBasedOnCharPos"